src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -949,6 +949,16 @@ theorem not_not {b : BitVec w} : ~~~(~~~b) = b := by
   ext i
   simp
 
+@[simp] theorem getMsb_not {x : BitVec w} :
+    (~~~x).getMsbD i = (decide (i < w) && !(x.getMsbD i)) := by
+  simp only [getMsbD]
+  by_cases h : i < w
+  · simp_all; omega
```

Comment:
You can use just `simp [h]` here. `simp_all` is a more powerful tool. It basically goes through all hypothesis `h_x` and does `simp [all_hyps_except h_x] at h_x` until a fixpoint.